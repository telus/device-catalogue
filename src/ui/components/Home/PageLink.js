import React from 'react';
import Link from '../Common/LinkComponent';
import TdsLink from '@tds/core-link';
import Card from '@tds/core-card';
import Box from '@tds/core-box';
import Image from '@tds/core-image';

const PageLink = ({ image, path, linkText, altText }) => {
  return (
    <Card variant="defaultOnlyBorder">
      <Box between={3} className="item-links">
        <Image src={image} alt={altText} width={90}/>
        <div>
          <Link TDSLink={TdsLink} to={path}>{linkText}</Link>
        </div>
      </Box>
    </Card>
  )
}

export default PageLink;