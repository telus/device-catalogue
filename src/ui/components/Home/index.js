import React from "react";
import FlexGrid from "@tds/core-flex-grid";
import Heading from "@tds/core-heading";
import Box from "@tds/core-box";
import Image from "@tds/core-image";
import Card from "@tds/core-card";
import ButtonLink from "@tds/core-button-link";
import A11yContent from "@tds/core-a11y-content";
import iphoneImage from "../../../../public/images/iphone.jpg";
import watchImage from "../../../../public/images/watch.jpg";

/**
 * Home page that shows two cards with links to go to watches/iphones page.
 * @returns
 */
const Home = () => {
  return (
    <FlexGrid>
      <Box vertical={6}>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={6}>
            <Card variant="defaultWithBorder">
              <FlexGrid.Row horizontalAlign="center">
                <Box between={2} vertical={3}>
                  <Heading level="h1">iPhones</Heading>
                </Box>
              </FlexGrid.Row>
              <FlexGrid.Row horizontalAlign="center">
                <Image
                  src={iphoneImage}
                  width={191}
                  height={238}
                  alt="Image of a watch"
                />
              </FlexGrid.Row>
              <FlexGrid.Row horizontalAlign="center">
                <Box vertical={5} between={1}>
                  <ButtonLink href="/iphones">
                    <span>
                      Shop <A11yContent>iPhones</A11yContent>Now
                    </span>
                  </ButtonLink>
                </Box>
              </FlexGrid.Row>
            </Card>
          </FlexGrid.Col>
          <FlexGrid.Col xs={12} md={6}>
            <Card variant="defaultWithBorder">
              <FlexGrid.Row horizontalAlign="center">
                <Box between={2} vertical={3}>
                  <Heading level="h1">Watches</Heading>
                </Box>
              </FlexGrid.Row>
              <FlexGrid.Row horizontalAlign="center">
                <Image
                  src={watchImage}
                  width={191}
                  height={238}
                  alt="Image of a watch"
                />
              </FlexGrid.Row>
              <FlexGrid.Row horizontalAlign="center">
                <Box vertical={5} between={1}>
                  <ButtonLink href="/watches">
                    <span>
                      Shop <A11yContent>watches</A11yContent>Now
                    </span>
                  </ButtonLink>
                </Box>
              </FlexGrid.Row>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </Box>
    </FlexGrid>
  );
};

export default Home;
