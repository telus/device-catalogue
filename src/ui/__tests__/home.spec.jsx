import React from "react";
import { render } from "enzyme";
import Home from "../components/Home";

describe("Device Card", () => {
  it("renders correctly with props", () => {
    const container = render(<Home />);

    expect(container.html()).toMatchSnapshot();
    expect(container.find("img").length).toEqual(2);
    expect(container.html().includes("a")).toEqual(true);
  });
});
