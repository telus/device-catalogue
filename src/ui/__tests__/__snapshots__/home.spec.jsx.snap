// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device Card renders correctly with props 1`] = `"<div class=\\"sc-bdfBQB jzdNVf\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 cbrPva\\"><div class=\\"Col__StyledCol-sc-15yvjc7-0 fPSiAN\\"><div class=\\"sc-bdfBQB cXMDiG sc-hKgJUU eocGKl sc-hKgJUU eocGKl\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 dkMqEl\\"><div class=\\"sc-bdfBQB hDaciT\\"><h1 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 wprAR\\">iPhones</h1></div></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 dkMqEl\\"><img src=\\"public/images/iphone.jpg\\" width=\\"191\\" height=\\"238\\" alt=\\"Image of a watch\\" class=\\"Image__StyledImage-b16p20-0 DCldW\\"></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 dkMqEl\\"><div class=\\"sc-bdfBQB izRkDg\\"><a href=\\"/iphones\\" class=\\"Button__StyledButton-sa7xo3-0 sc-jSgvzq fPEGeE klvbyh\\"><span>Shop <span class=\\"A11yContent__StyledA11yContent-sc-15wexwk-0 jolBxv\\">iPhones</span>Now</span></a></div></div></div></div><div class=\\"Col__StyledCol-sc-15yvjc7-0 fPSiAN\\"><div class=\\"sc-bdfBQB cXMDiG sc-hKgJUU eocGKl sc-hKgJUU eocGKl\\"><div class=\\"Row__StyledRow-sc-1goz0ht-0 dkMqEl\\"><div class=\\"sc-bdfBQB hDaciT\\"><h1 data-testid=\\"heading\\" class=\\"Heading__StyledHeading-mpz92r-0 wprAR\\">Watches</h1></div></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 dkMqEl\\"><img src=\\"public/images/watch.jpg\\" width=\\"191\\" height=\\"238\\" alt=\\"Image of a watch\\" class=\\"Image__StyledImage-b16p20-0 DCldW\\"></div><div class=\\"Row__StyledRow-sc-1goz0ht-0 dkMqEl\\"><div class=\\"sc-bdfBQB izRkDg\\"><a href=\\"/watches\\" class=\\"Button__StyledButton-sa7xo3-0 sc-jSgvzq fPEGeE klvbyh\\"><span>Shop <span class=\\"A11yContent__StyledA11yContent-sc-15wexwk-0 jolBxv\\">watches</span>Now</span></a></div></div></div></div></div></div>"`;
